DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -60,17 +71,78 @@ string userIp
             if (!delegateRegistrationModel.Approved)
             {
                 var contactInfo = centresDataService.GetCentreManagerDetails(delegateRegistrationModel.Centre);
-                var approvalEmail = GenerateApprovalEmail(contactInfo.email, contactInfo.firstName,
+                var approvalEmail = GenerateApprovalEmail(
+                    contactInfo.email,
+                    contactInfo.firstName,
                     delegateRegistrationModel.FirstName,
-                    delegateRegistrationModel.LastName, baseUrl);
+                    delegateRegistrationModel.LastName,
+                    baseUrl
+                );
                 emailService.SendEmail(approvalEmail);
             }
 
             return (candidateNumber, delegateRegistrationModel.Approved);
         }
 
-        private Email GenerateApprovalEmail
-        (
+        public string RegisterAdminDelegate(
+            RegistrationModel registrationModel
+        )
+        {
+            var delegateRegistrationModel = new DelegateRegistrationModel(
+                registrationModel.FirstName,
+                registrationModel.LastName,
+                registrationModel.Email,
+                registrationModel.Centre,
+                registrationModel.JobGroup,
+                registrationModel.PasswordHash,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null
+            );
+            delegateRegistrationModel.Approved = true;
+            var candidateNumber = registrationDataService.RegisterDelegate(delegateRegistrationModel);
+            if (candidateNumber == "-1" || candidateNumber == "-4")
+            {
+                return candidateNumber;
+            }
+
+            passwordDataService.SetPasswordByCandidateNumber(candidateNumber, delegateRegistrationModel.PasswordHash);
+
+            return candidateNumber;
+        }
+
+        public bool RegisterCentreManager(
+            RegistrationModel registrationModel
+        )
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                var candidateNumber = RegisterAdminDelegate(registrationModel);
+                if (candidateNumber == "-1")
+                {
+                    throw new Exception("Delegate account could not be created");
+                }
+
+                registrationDataService.RegisterCentreManagerAdmin(registrationModel);
+
+                centresDataService.SetCentreAutoRegistered(registrationModel.Centre);

```

Comment:
I know that is what the ticket says, but does this not affect any subsequent delegate registrations? Seems a bit weird to change properties of the centre when registering an admin.

Reply:
We need to set AutoRegistered to true to indicate that the centre manager has registered. I don't quite follow how this is unexpected or affects delegate registrations?

Reply:
Huh I guess it is just a misleading column name then. Really it means HasManagerRegistered. From the naming I assumed it meant something to do with allowing automatic future registrations.